File: prostudy-api/src/main/java/com/planbtech/prostudy/entities/model/Question.java
package com.planbtech.prostudy.entities.model;

import jakarta.persistence.*;
import lombok.*;

import java.util.List;

@Entity
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Question {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "questionid")
    private Long question;

    @Column(name = "questiontext", columnDefinition = "TinyText")
    private String questionText;

    @Column(name = "questionimage", columnDefinition = "TinyText")
    private String quesyionImage;

    @ManyToOne
    @JoinColumn(name = "testid", nullable = false)
    private SkillTest testId;

    @OneToMany(mappedBy = "question")
    private List<Answer> answerList;
}

File: PlanBTech/src/app/class/question.ts
import { Answer } from "./answer";
import { Test } from "./test";

export class Question {

    questionId:Number;
    questionText:String;
    testeId:Test;
    answers:Answer[];

    constructor(){
        this.questionId     = 0;
        this.questionText   = '';    
        this.testeId        = new Test();
        this.answers        = [];
      }
}
